src/main/java/online/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,165 @@
+package online.shop.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import online.shop.dao.OrderDao;
+import online.shop.exception.DataProcessingException;
+import online.shop.lib.Dao;
+import online.shop.model.Order;
+import online.shop.model.Product;
+import online.shop.util.ConnectionUtil;
+
+@Dao
+public class OrderDaoJdbcImpl implements OrderDao {
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        List<Order> orders = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM orders WHERE user_id = ?";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                orders.add(getOrderFromResultSet(resultSet));
+            }
+            return orders;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get order by user id : " + userId, e);
+        }
+    }
+
+    @Override
+    public Order create(Order order) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "INSERT INTO orders(user_id) value (?)";
+            PreparedStatement statement =
+                    connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                order.setId(resultSet.getLong(1));
+            }
+            addProductsToOrder(order);
```

Comment:
You open new connection inside existing connection. Checklist (rule 10) says to avoid this.
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_23.html

Try to move ````addProductsToOrder(order);```` outside of try-catch block.